wdl/qc_imputation.wdl

```diff
@@ -23,44 +22,64 @@ workflow qc_imputation {
     #
     ## ... Maybe better remove symmetric MAF > 0.4 etc. at some point? Mitja
 
-    # run joint qc and panel comparison per chromosome
+    scatter (i in range(length(vcfs))) {
+        # convert batches from vcf to bed
+        call vcf_to_bed {
+            input: vcf=vcfs[i], exclude_samples=exclude_samples[i], include_regex=include_regex, docker=docker
+        }
+        # run glm between each batch and panel
+        call glm_vs_panel {
+            input: base=sub(sub(basename(vcfs[i], ".vcf"), "_original", ""), ".calls", ""),
+            beds=[vcf_to_bed.out_bed], bims=[vcf_to_bed.out_bim], fams=[vcf_to_bed.out_fam],
+            panel_bed=panel_bed, high_ld_regions=high_ld_regions, docker=docker
+        }
+        # get variants to exclude based on missingness
+        call missingness {
+            input: base=sub(sub(basename(vcfs[i], ".vcf"), "_original", ""), ".calls", ""),
+            bed=vcf_to_bed.out_bed, bim=vcf_to_bed.out_bim, fam=vcf_to_bed.out_fam, docker=docker
+        }
+    }
+
+    # run snpstats per chromosome
     scatter (chr in chrs) {
-        call qc_sub.chr_qc as chr_qc {
-            input: exclude_samples_loc=exclude_samples_loc,
-                vcf_loc=vcf_loc,name=name, chr=chr, panel_vcf=ref_panel[chr], include_regex=include_regex,
-                Fs=f,genome_build=genome_build, high_ld_regions=high_ld_regions, docker=docker
+        call snpstats {
+            input: beds=vcf_to_bed.out_bed, bims=vcf_to_bed.out_bim, fams=vcf_to_bed.out_fam,
+            chr=chr, outname=name+"_chr"+chr, Fs=f, genome_build=genome_build, docker=docker
         }
     }
+
+    # gather snpstats and create a list of variants to be excluded from all batches
+    call gather_snpstats_joint_qc {
+        input: name=name, snpstats=snpstats.snpstats,
+        exclude_variants_batch_nonpass=vcf_to_bed.exclude_variants,
+        exclude_variants_batch_missingness=missingness.exclude_variants,
+        docker=docker
+    }
+
     # run qc per batch across all chromosomes
     scatter (i in range(length(vcfs))) {
-        String base1 = sub(sub(basename(vcfs[i], ".vcf"), "_original", ""), ".calls", "")
-        # combine panel comparison across chrs and create plot
+        String base_batch = sub(sub(basename(vcfs[i], ".vcf"), "_original", ""), ".calls", "")
+        # get variants to exclude based on panel comparison and create plot
         call panel_comparison {
-            input: base=base1, freq=transpose(chr_qc.freq)[i], freq_panel=transpose(chr_qc.freq_panel)[i],
-            files_glm=transpose(chr_qc.glm)[i], exclude_variants_panel=transpose(chr_qc.exclude_variants_panel)[i],
-            docker=docker
+            input: base=base_batch, dataset_freq_panel_intersect=glm_vs_panel.dataset_freq_panel_intersect[i],
+            freq_panel=sub(panel_bed, ".bed$", ".afreq"), files_glm=[glm_vs_panel.glm[i]], bim=vcf_to_bed.out_bim[i], docker=docker
```

Comment:
here the same cosmetic comment as previously of passing a list where only a single file is actually needed.